documentation/specs/TabPanels.md

```diff
@@ -0,0 +1,163 @@
+
+  /**
+   * Whether tabs are activated automatically on focus or manually.
+   *
+   * @default "automatic"
+   */
+  keyboardActivation?: "automatic" | "manual";
+
+  /**
+   * Handler that is called when the selection changes.
+   */
+  onSelectionChange?: (key: Key) => void;
+
+  /**
+   * The currently selected key in the collection (controlled).
+   */
+  selectedKey?: Key | null;
+};
+
+type TabPanelsTabs = {
+  /**
+   * The tab items to display. Item keys should match the key of the
+   * corresponding <Item> within the <TabPanels.Panels> element.
+   */
+  children: ReactNode;
+}
+
+type TabPanelsPanels = {
+  /**
+   * The contents of each tab. Item keys should match the key of the
+   * corresponding <Item> within the <TabPanels.Tabs> element.
+   */
+  children: ReactNode;
+}
+```
+
+### Example Usage
+
+_Basic usage_:
+
+```tsx
+import { TabPanels } from "@easypost/easy-ui/TabPanels";
+
+function BasicTabPanels() {
+  return (
+    <TabPanels
+      aria-label="History of Ancient Rome"
+      // managing selection (uncontrolled)
+      defaultSelectedKey="one"
+      // managing selection (controlled)
+      selectedKey="one"
+      onSelectionChange={(key) => {}}
+      // keyboard activation can either be automatic or manual
+      keyboardActivation="manual"
+    >
+      <TabPanels.Tabs>
+        <Item key="one">Founding of Rome</Item>
+        <Item key="two">Monarchy and Republic</Item>
+        <Item key="three">Empire</Item>
+      </TabPanels.Tabs>
+      <TabPanels.Panels>
+        <Item key="one">Arma virumque cano, Troiae qui primus ab oris.</Item>
+        <Item key="two">Senatus Populusque Romanus.</Item>
+        <Item key="three">Alea jacta est.</Item>
+      </TabPanels.Panels>
+    </TabPanels>
+  );
+}
+```
+
+_Custom layout_:
+
+`TabPanels.Tabs` and `TabPanels.Panels` are not required to be the immediate children of `TabPanels`. If you need additional structure in-between, `TabPanels` supports this.
+
+```tsx
+import { TabPanels } from "@easypost/easy-ui/TabPanels";
+
+function CustomLayoutTabPanels() {
+  return (
+    <TabPanels aria-label="History of Ancient Rome">
+      <div>
+        <div>
+          <TabPanels.Tabs>
+            <Item key="one">Founding of Rome</Item>
+            <Item key="two">Monarchy and Republic</Item>
+            <Item key="three">Empire</Item>
+          </TabPanels.Tabs>
+        </div>
+        <div>
+          <TabPanels.Panels>
+            <Item key="one">Arma virumque cano, Troiae qui primus ab oris.</Item>
+            <Item key="two">Senatus Populusque Romanus.</Item>
+            <Item key="three">Alea jacta est.</Item>
+          </TabPanels.Panels>
+        </div>
+      </div>
+    </TabPanels>
+  );
+}
+```
+
+---
+
+## Behavior
+
+### Accessibility
+
+- A tab should never contain an interactive element, as doing so makes it difficult for users of assistive technologies to focus or click on the tabs. Interactive elements may live within the panels.
```

Comment:
**Thought:** Curious if there is an eslint plugin that could enforce this or maybe we could write one

Reply:
good thought. i'll look into that